server/kv.go

```diff
@@ -104,19 +106,10 @@ func (store store) get(key string, v interface{}) (returnErr error) {
 		returnErr = errors.WithMessage(returnErr, "failed to get from store")
 	}()
 
-	data, appErr := store.plugin.API.KVGet(key)
-	if appErr != nil {
-		return appErr
-	}
-	if data == nil {
-		return kvstore.ErrNotFound
-	}
-
-	err := json.Unmarshal(data, v)
+	err := store.plugin.client.KV.Get(key, &v)
```

Comment:
Previously, `v` was passed to `json.Unmarshal` meaning it had to be an pointer. Why does a pointer to `v` get passed now?